test/test_quantizer.py

```diff
@@ -10,58 +10,131 @@
+
+Arguments:
+    name: Key name in the stats dictionary
+wrapper
+    value_stats: Stats dict from observer wrapper
+
+
+Output:
+    scale, zero_point
+"""
+
+
+def calcQParamFunc(name, value_stats):
+    scaleT = 2.0 * (torch.max(value_stats[name][1],
+                    -value_stats[name][0]) / 255.0)
+    scale = scaleT.item()
+    zero_point = 0
+    return scale, zero_point
+
+
+r"""
+ Unified Dictionary for all qparam
+"""
+
+
+def getAllQParamDict(allqparam_dict, quantObj):
+    if allqparam_dict is None:
+        allqparam_dict = {}
+    qparam_dict = quantObj.getQParamDict()
+    if qparam_dict is None:
+        return
+    allqparam_dict.update(qparam_dict)
+
+
+r"""
+This is an example QuantConfigTemplate which will be used to collect
 stats across batches by running torch script/trace module, from the
 observer nodes inserted in the graph. These stats are used to compute
 Quantization Parameters. These will be passed to quantizer to be used
 as arguments for quant ops in quantization pass.
 """
 
 
-class ObserverModule:
-    def __init__(self, userObserver=None, userCalcQParam=None):
+class QuantConfigTemplate:
+    def __init__(self, tensorType=None, observerImpl=None, calcQParamImpl=None):
         self.value_stats = {}
         self.qparam_dict = {}
         self.averaging_constant = 0.001
-        if userObserver is not None:
-            self.observerImpl = userObserver
-        if userCalcQParam is not None:
-            self.calcQParamImpl = userCalcQParam
+        self.observerImpl = observerImpl
+        self.calcQParamImpl = calcQParamImpl
+        self.tensor_type = tensorType
 
     def resetStats(self):
         self.value_stats = {}
         return
 
-    def observerImpl(self, value, stats, avgconstant):
-        # Default observer. Stats needed for accumulation
-        stats[0] = torch.min(value)
-        stats[1] = torch.max(value)
-        return value
-
     def observer(self, value, name):
+        if self.observerImpl is None:
+            return
         if name not in self.value_stats:
             self.value_stats[name] = []
             stats = torch.zeros(2)
         else:
             stats = self.value_stats[name]
-        self.observerImpl(value, stats, self.averaging_constant)
+        self.observerImpl(value, stats)
         self.value_stats.update({name: stats})
         return value
 
-    def calcQParamImpl(self, qparam_dict, value_stats):
-        # Test QParam computation. Used by default for test
-        for name in value_stats:
-            qparam = torch.zeros(2)
-            scaleT = value_stats[name][0]
-            scale = scaleT.item()
-            zero_pointT = value_stats[name][1]
-            zero_point = zero_pointT.item()
-            qparam_dict.update({name: (scale, zero_point)})
-
     def calcQParam(self):
         self.qparam_dict = {}
-        self.calcQParamImpl(self.qparam_dict, self.value_stats)
+        if self.calcQParamImpl is None:
```

Comment:
It's hard to understand what's a meaningful change and what's just a code movement here - please seaparate those two into separate PRs.